### DISCUSSION TITLE
How do I prevent network switching on connect if the wallet has already set a reliable network?

### DISCUSSION THREAD

**Discussion by chrisbrights** [family/connectkit#225] [Q&A] [answered]
Hello, 

Thanks for your hard work.
I am now working on DApp Development and using your ConnectKit to connect wallets.
For the test purpose, I want to use both Goerli and BscTestnet, so I wrote the below code.

```
// create client to use goerli and bscTestnet
const client = createClient(
  getDefaultClient({
    appName: 'Iniverse',
    alchemyId:  process.env.NEXT_PUBLIC_ALCHEMY_GOERLI_KEY,
    walletConnectProjectId: process.env.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID,
    chains: [goerli, bscTestnet],
  })
);

// using connect button
<ConnectKitButton />
```
The problem is that a ConnectKitButton click makes Goerli switch even if I already set BscTestnet on Metamask.
I want to prevent network switching if the wallet network is already set as Goerli or BscTestnet, but I'm not sure how to handle this problem when I use your ConnectKit.
Would you please tell me how to do this?

**Comment by lochie** [comment, marked as answer]
@chrisbrights you can disable this behaviour by setting `enforceSupportedChains` to `false`
https://docs.family.co/connectkit/api-reference#enforceSupportedChains

**Reply by chrisbrights** [reply]
Hi Lochie, 

Nice to meet you.
I am afraid you misunderstood something. 
If I set enforceSupportedChains to false, it won't perform network switch if wallet network is already set as Sepolia too.
I want the network switch to Goerli happens if wallet network is neither Goerli nor BscTestnet while not happening if wallet network is Goerli or BscTestnet.
Would you tell me what you think about this problem?

**Reply by lochie** [reply]
Oops sorry! You can turn off chain targeting by setting `initialChainId` to `0`
https://docs.family.co/connectkit/api-reference#initialChainId

**Reply by chrisbrights** [reply]
Yay! 🏆 it works. 🙌 
Thanks for your kind help!!!!
